examples/testgputext/SDL_math3d.h

```diff
@@ -0,0 +1,309 @@
+
+    if (mag == 0) {
+        return (SDL_Vec3) {0, 0, 0};
+    } else if (mag == 1){
+        return vec;
+    } else {
+        return (SDL_Vec3) {vec.x/mag, vec.y/mag, vec.z/mag};
+    }
+}
+
+static inline SDL_Vec3 SDL_Vec3Add(SDL_Vec3 vec1, SDL_Vec3 vec2)
+{
+    return SDL_Vector3(vec1.x + vec2.x, vec1.y + vec2.y, vec1.z + vec2.z);
+}
+
+static inline SDL_Vec3 SDL_Vec3Sub(SDL_Vec3 vec1, SDL_Vec3 vec2)
+{
+    return SDL_Vector3(vec1.x - vec2.x, vec1.y - vec2.y, vec1.z - vec2.z);
+}
+
+static inline SDL_Vec3 SDL_Vec3MultiplyFloat(SDL_Vec3 vec, float val)
+{
+    return SDL_Vector3(vec.x * val, vec.y * val, vec.z * val);
+}
+
+static inline float SDL_Vec3Dot(SDL_Vec3 vec1, SDL_Vec3 vec2)
+{
+    return (vec1.x * vec2.x + vec1.y * vec2.y + vec1.y * vec2.y);
+}
+
+static inline SDL_Vec3 SDL_Vec3Cross(SDL_Vec3 vec1, SDL_Vec3 vec2) {
+    return SDL_Vector3(
+        vec1.y * vec2.z - vec1.z * vec2.y,
+        vec1.z * vec2.x - vec1.x * vec2.z,
+        vec1.x * vec2.y - vec1.y * vec2.x
+    );
+}
+
+
+static inline SDL_Mat4X4 SDL_Matrix4X4(
+    float m00, float m10, float m20, float m30,
+    float m01, float m11, float m21, float m31,
+    float m02, float m12, float m22, float m32,
+    float m03, float m13, float m23, float m33
+) {
+    return (SDL_Mat4X4) {
+        .m[0][0] = m00, .m[1][0] = m10, .m[2][0] = m20, .m[3][0] = m30,
+        .m[0][1] = m01, .m[1][1] = m11, .m[2][1] = m21, .m[3][1] = m31,
+        .m[0][2] = m02, .m[1][2] = m12, .m[2][2] = m22, .m[3][2] = m32,
+        .m[0][3] = m03, .m[1][3] = m13, .m[2][3] = m23, .m[3][3] = m33
+    };
+}
+
+static inline SDL_Mat4X4 SDL_MatrixIdentity(void)
+{
+    return SDL_Matrix4X4(
+        1,  0,  0,  0,
+        0,  1,  0,  0,
+        0,  0,  1,  0,
+        0,  0,  0,  1
+    );
+}
+
+static inline SDL_Mat4X4 SDL_MatrixTranspose(SDL_Mat4X4 mat)
+{
+    SDL_Mat4X4 res;
+
+    for (int i = 0; i < 4; i++) {
+        for (int j = 0; j < 4; j++) {
+            res.m[i][j] = mat.m[i][j];
+        }
+    }
+
+    return res;
+}
+
+static inline SDL_Mat4X4 SDL_MatrixMultiply(SDL_Mat4X4 mat1, SDL_Mat4X4 mat2)
+{
+    SDL_Mat4X4 res;
+
+    for (int i = 0; i < 4; i++) {
+        for (int j = 0; j < 4; j++) {
+            float sum = 0;
+            for (int x = 0; x < 4; x++) {
+                sum += mat1.m[x][j] * mat2.m[i][x];
+            }
+            res.m[i][j] = sum;
+        }
+    }
+    
+    return res;
+}
+
+static inline SDL_Mat4X4 SDL_MatrixScaling(SDL_Vec3 scale)
+{
+    float x = scale.x, y = scale.y, z = scale.z;
+    return SDL_Matrix4X4(
+         x,  0,  0,  0,
+         0,  y,  0,  0,
+         0,  0,  z,  0,
+         0,  0,  0,  1
+    );
+}
+
+static inline SDL_Mat4X4 SDL_MatrixTranslation(SDL_Vec3 offset)
+{
+    return SDL_Matrix4X4(
+         1,  0,  0,  offset.x,
+         0,  1,  0,  offset.y,
+         0,  0,  1,  offset.z,
+         0,  0,  0,         1
+    );
+}
+
+static inline SDL_Mat4X4 SDL_MatrixRotationX(float angle)
```

Comment:
I believe this should be instead:

```suggestion
static inline SDL_Mat4X4 SDL_MatrixRotationX(float angle)
{
    float cos = SDL_cos(angle);
    float sin = SDL_sin(angle);

    return SDL_Matrix4X4(
         1,  0,    0,  0,
         0, cos, -sin,  0,
         0, sin,  cos,  0,
         0,  0,    0,  1
    );
}
```
At the moment this function is a copy of `SDL_MatrixRotationY`.